app.js

```diff
@@ -232,8 +233,59 @@ async function setFinalReferenceValue(poolId, geostatsData) {
             return "Error while doing the transaction."
         })
 
+}
+
+function isValidRequest(requestData, poolExpiryTime, poolCreationTime) {
+
+    agg_x = requestData.agg_x
+    dataset = requestData.dataset
+    band = requestData.band
+    scale = requestData.scale
+    start_date_string = requestData.start_date
+    end_date_string = requestData.end_date
+    geometry = requestData.geometry
+
+    pool_expiry_date_string = (new Date(parseInt(poolExpiryTime) * 1000)).toISOString().split('T')[0];
+    pool_creation_date_string = (new Date(parseInt(poolCreationTime) * 1000)).toISOString().split('T')[0];
+
+    if (agg_x == "agg_mean" || agg_x == "agg_max" || agg_x == "agg_min" || agg_x == "agg_median" || agg_x == "agg_stdDev" || agg_x == "agg_variance") {
+        if (dataset == "VIIRS_VI" && band == "NDVI" && scale == "1000") {
+            today = new Date()
+            // tenDaysAgo = new Date(new Date().setDate(today.getDate() - 10))
+            start_date = new Date(start_date_string)
+            end_date = new Date(end_date_string)
+            pool_creation_date = new Date(pool_creation_date_string)
+
+            diffTime = Math.abs(end_date - start_date);
+            diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24)); 
+
+            if (diffDays >= 30 && end_date_string == pool_expiry_date_string && start_date > pool_creation_date) {
```

Comment:
I think this is not the condition we eventually agreed on. See here: https://github.com/shambadynamic/Shamba-Diva-Middleware/pull/2#discussion_r1041908542